api/envoy/extensions/filters/listener/proxy_protocol/v3/proxy_protocol.proto

```diff
@@ -17,4 +18,57 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 message ProxyProtocol {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.listener.proxy_protocol.v2.ProxyProtocol";
+
+  enum ValueType { STRING = 0; }
+
+  // ValueEncode defines the encoding algorithm.
+  enum ValueEncode {
+    // The value is not encoded.
+    NONE = 0;
+  }
+
+  // [#next-free-field: 6]
+  message KeyValuePair {
+    // The namespace — if this is empty, the filter's namespace will be used.
+    string metadata_namespace = 1;
+
+    // The key to use within the namespace.
+    string key = 2 [(validate.rules).string = {min_bytes: 1}];
+
+    // The value to pair with the given key.
+    //
+    // When used for a `on_header_present` case, if value is non-empty it'll be used
+    // instead of the header value. If both are empty, no metadata is added.
+    //
+    // When used for a `on_header_missing` case, a non-empty value must be provided
+    // otherwise no metadata is added.
+    //string value = 3;
```

Comment:
Removed the value field. The ValueType and ValueEncode are removed as well since their purpose are supporting the value field. In the future they can be added together if the function is needed.

Reply:
The `ValueType` and `ValueEncode` still could make sense, because there is still a value (the data that came in Proxy protocol header.

However, given that they're currently unimplemented (enum with only a single value), I think it's fine to leave them removed. If something like that is needed in the future, it's easy to add them in a backwards-compatible way, as long as the default is "no encoding, type string", so let's leave it as you have it now.